modules/nvidia-gpu-gcp-adding-a-gpu-node.adoc

```diff
@@ -0,0 +1,322 @@
+myclustername-2pt9p-worker-c-6pbg6       Running   n2-standard-4   us-central1   us-central1-c   8h
+----
+
+. Make a copy of one of the existing compute `MachineSet` definitions and output the result to a JSON file by running the following command. This will be the basis for the GPU-enabled compute machine set definition.
++
+[source,terminal]
+----
+$ oc get machineset myclustername-2pt9p-worker-a -n openshift-machine-api -o json  > <output_file.json>
+----
+
+. Edit the JSON file and make the following changes to the new `MachineSet` definition:
++
+* Rename the machine set `name` by inserting the substring `gpu` in `metadata.name` and in both instances of `machine.openshift.io/cluster-api-machineset`.
+* Change the `machineType` of the new `MachineSet` definition to `a2-highgpu-1g`, which includes an NVIDIA A100 GPU.
++
+[source,terminal]
+----
+jq .spec.template.spec.providerSpec.value.machineType ocp_4.12_machineset-a2-highgpu-1g.json
+
+"a2-highgpu-1g"
+----
++
+The `<output_file.json>` file is saved as `ocp_4.12_machineset-a2-highgpu-1g.json`.
+
+. Update the following fields in `ocp_4.12_machineset-a2-highgpu-1g.json`:
++
+* `.metadata.name` to a name containing `gpu`.
+
+* `.spec.selector.matchLabels["machine.openshift.io/cluster-api-machineset"]` to
+match the new `.metadata.name`.
+
+* `.spec.template.metadata.labels["machine.openshift.io/cluster-api-machineset"]`
+to match the new `.metadata.name`.
+
+* `.spec.template.spec.providerSpec.value.MachineType` to `a2-highgpu-1g`.
+
+* Add the following line under `machineType`: `"onHostMaintenance": "Terminate".
++
+These lines now look like this:
++
+[source,json]
+----
+"machineType": "a2-highgpu-1g",
+"onHostMaintenance": "Terminate",
+----
+
+. To verify your changes, perform a `diff` of the original compute definition and the new GPU-enabled node definition by running the following command:
++
+[source,terminal]
+----
+$ oc get machineset/myclustername-2pt9p-worker-a -n openshift-machine-api -o json | diff ocp_4.12_machineset-a2-highgpu-1g.json -
+----
++
+.Example output
++
+[source,terminal]
+----
+15c15
+<         "name": "myclustername-2pt9p-worker-gpu-a",
+---
+>         "name": "myclustername-2pt9p-worker-a",
+25c25
+<                 "machine.openshift.io/cluster-api-machineset": "myclustername-2pt9p-worker-gpu-a"
+---
+>                 "machine.openshift.io/cluster-api-machineset": "myclustername-2pt9p-worker-a"
+34c34
+<                     "machine.openshift.io/cluster-api-machineset": "myclustername-2pt9p-worker-gpu-a"
+---
+>                     "machine.openshift.io/cluster-api-machineset": "myclustername-2pt9p-worker-a"
+59,60c59
+<                         "machineType": "a2-highgpu-1g",
+<                         "onHostMaintenance": "Terminate",
+---
+>                         "machineType": "n2-standard-4",
+----
+
+. Create the GPU-enabled compute machine set from the definition by running the following command:
```

Comment:
May just be my unfamiliarity with JSON but does "definition" need more identification here? (e.g., "from the definition file")

Reply:
Added "file"